[IMP] ai: add documentation about pgvector (#17852)
odoo/documentation: content/administration/on_premise/source.rst

         Because the PostgreSQL user has the same name as the Unix login, it is possible to connect
         to the database without a password.

.. note::
   If you want to use **Odoo's AI features**, the `pg-vector` PostgreSQL extension is required.

   Note that `pg-vector` is available only for PostgreSQL version 15 and up.

   .. tabs::

      .. group-tab:: Linux

         #. Navigate to your temporary folder

            .. code-block:: console

               $ cd /tmp

         #. Clone the `pg-vector` `GitHub repository <https://github.com/pgvector/pgvector>`_:

            .. code-block:: console

               $ git clone https://github.com/pgvector/pgvector.git

         #. Navigate into the directory:

            .. code-block:: console

               $ cd pgvector

         #. Compile the extension:

            .. code-block:: console

               $ make

         #. Install the extension:

            .. code-block:: console

               $ sudo make install

>          If you run into any issues during installation, make sure to check the official `installation notes <https://github.com/pgvector/pgvector#installation-notes---linux-and-mac>`__

Reviewer: ```suggestion
         If you run into any issues during installation, make sure to check the official `installation notes <https://github.com/pgvector/pgvector#installation-notes---linux-and-mac>`__.
```

      .. group-tab:: Mac OS

         #. Navigate to your temporary folder

            .. code-block:: console

               $ cd /tmp

         #. Clone the `pg-vector` `GitHub repository <https://github.com/pgvector/pgvector>`_:

            .. code-block:: console

               $ git clone https://github.com/pgvector/pgvector.git

         #. Navigate into the directory:

            .. code-block:: console

               $ cd pgvector

         #. Compile the extension:

            .. code-block:: console

               $ make

         #. Install the extension:

            .. code-block:: console

               $ sudo make install

>          If you run into any issues during installation, make sure to check the official `installation notes <https://github.com/pgvector/pgvector#installation-notes---linux-and-mac>`__

Reviewer: ```suggestion
         If you run into any issues during installation, make sure to check the official `installation notes <https://github.com/pgvector/pgvector#installation-notes---linux-and-mac>`__.
```

      .. group-tab:: Windows

         #. Navigate to your temporary folder

            .. code-block:: console

               $ cd %TEMP%

         #. Clone the `pg-vector` `GitHub repository <https://github.com/pgvector/pgvector>`_:

            .. code-block:: console

               $ git clone https://github.com/pgvector/pgvector.git

         #. Navigate into the directory:

            .. code-block:: console

               $ cd pgvector

         #. Ensure C++ support in Visual Studio is installed and run `x64 Native Tools Command Prompt for VS [version]` as administrator.

         #. Set your `PGROOT` environment variable:

            .. code-block:: console

               $ set "PGROOT=C:\Program Files\PostgreSQL\[Your PostgreSQL Version]"

         #. Compile the extension:

            .. code-block:: console

               $ nmake /F Makefile.win

         #. Install the extension:

            .. code-block:: console

               $ nmake /F Makefile.win install

>          If you run into any issues during installation, make sure to check the official `installation notes <https://github.com/pgvector/pgvector#installation-notes---windows>`__

Reviewer: ```suggestion
         If you run into any issues during installation, make sure to check the official `installation notes <https://github.com/pgvector/pgvector#installation-notes---windows>`__.
```

               $ sudo npm install -g rtlcss

> .. note::

Reviewer: Move this note in the PostgreSQL section.

   If you want to use **Odoo's AI features**, the `pg-vector` PostgreSQL extension is required.

>    Note that `pg-vector` is available only for postgres version 15 and up.

Reviewer: Single line returns don't do anything in RST. Make it a separate paragraph:

```suggestion
   If you want to use **Odoo's AI features**, the `pg-vector` PostgreSQL extension is required.
   
   Note that `pg-vector` is available only for postgres version 15 and up.
```

>    Note that `pg-vector` is available only for postgres version 15 and up.

Reviewer: ```suggestion
   If you want to use **Odoo's AI features**, the `pg-vector` PostgreSQL extension is required.
   Note that `pg-vector` is available only for PostgreSQL version 15 and up.
```

   #. Navigate to your temporary folder

>       .. tabs::

Reviewer: Rather than split the instructions at the steps level, which is a bit messy, move your tabs above the numbered list for a cleaner structure.

>          .. group-tab:: Linux and macOS

Reviewer: Although this works, it would be better to keep separate Linux and macOS tabs in the context of this page, because tabs are synchronized by title, and users won't have to click tab headers in every section they read.

            .. code-block:: console

               $ cd /tmp

         .. group-tab:: Windows

            .. code-block:: console

               $ cd %TEMP%

   #. Clone the `pg-vector` `GitHub repository <https://github.com/pgvector/pgvector>`_:

      .. code-block:: console

         $ git clone https://github.com/pgvector/pgvector.git

   #. Navigate into the directory:

      .. code-block:: console

         $ cd pgvector

   .. tabs::

      .. group-tab:: Linux and macOS

>          If you run into any issues during installation, make sure to check the official `installation notes <https://github.com/pgvector/pgvector#installation-notes---linux-and-mac>`_

Reviewer: This should rather be moved to the end of the Linux and macOS tabs' content.

         #. Compile the extension:

            .. code-block:: console

               $ make

         #. Install the extension:

            .. code-block:: console

               $ sudo make install

      .. group-tab:: Windows

>          Ensure C++ support in Visual Studio is installed and `run x64 Native Tools Command Prompt for VS [version]` as administrator

Reviewer: This sounds like a numbered step on its own.
